walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +272,144 @@ public void notEnoughRoom() {
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void optionalHasNotEnoughRoom() {
+    // Have two attendees and one optional attendee, but make it so that the optional attendee
+    // doesn't have enough room but the two attendees do have space to
+    // have the meeting.
+    //
+    // Events  : |--A--| |-(A)-||---B--|
+    // Day     : |---------------------|
+    // Options :       |--------|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TimeRange.END_OF_DAY, true),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 3", TimeRange.fromStartEnd(TIME_0830AM, TIME_0845AM, true),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_B);
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+
+  @Test
+  public void optionalsHaveOption() {
+    // Have two optional attendees, but make it so that there is just enough room at one point in the day to
+    // have the meeting.
+    //
+    // Events  : |-(A)-|     |---(B)---|
+    // Day     : |---------------------|
+    // Options :       |-----|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TIME_0930AM, false),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(Arrays.asList(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES), 
+          TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void optionalsHaveNoOption() {
+    // Have two optional attendees, but make it so that there is no time during the day to
+    // have the meeting.
+    //
+    // Events  : |----(A)---||---(B)---|
+    // Day     : |---------------------|
+    // Options :      
+
+    Collection<Event> events = Arrays.asList(
```

Comment:
nit: typically you'll want exactly three code blocks in your test, separated by one line of whitespace: Arrange, Act, Assert. Also you'll want the description of the test case in comments (which is AWESOME) to be outside of the test body.

So in this case that would be: 

```
    // Have two optional attendees, but make it so that there is no time during the day to
    // have the meeting.
    //
    // Events  : |----(A)---||---(B)---|
    // Day     : |---------------------|
    // Options :      
@Test
public void optionalsHaveNoOption() {
  Collection<Event> events = Arrays.asList(
       new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0900AM, true),
            Arrays.asList(PERSON_A)),
        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TimeRange.END_OF_DAY, true),
            Arrays.asList(PERSON_B)));
    MeetingRequest request = new MeetingRequest(Arrays.asList(), DURATION_30_MINUTES);
    request.addOptionalAttendee(PERSON_A);
    request.addOptionalAttendee(PERSON_B);

    Collection<TimeRange> actual = query.query(events, request);

    Collection<TimeRange> expected = Arrays.asList();
    Assert.assertEquals(expected, actual);
}
```
Please use this format throughout.